DoCentre/DoCentre
# Add API documentations

# What's Changed?

Added API documentation for the currently implemented APIs:

- CreateDocument
- UserCreate
- UserLogin
- GetUsersByUsername

The definition of request and response structs is necessary for Swagger to recognize the JSON structure. We need them to annotate the Swagger comment.

This also affects the API implementation, but I believe making the request and response structs explicit has only advantages.

> [!note]
> - Two `XXX` comments are added for JSON fields that are also null/empty in the response body. I suggest removing them in the further pull request as these fields are ill-formed. The front-end is expected to check the status code and not access these fields.
> - The `GetUsersByUsername` endpoint returns `200` even on errors, which seems to be a mistake. If there's no user with the given username, the `users` list is simply empty and no error is returned. The error indicates other kinds of situations.

# How to Verify?

Start the server with `docker compose up` and visit `http://localhost:8080/swagger/index.html`. The documentation of the APIs should be shown properly.
